main.js

```diff
@@ -1 +1,145 @@
-  
+ /*
+ * Created by Jenya on 13.06.15.
+ */
+function onScreen (text) {
+    document.getElementById("p1").innerHTML += "<br>" + text;
+}
+
+var textLoaded="";
+var sourceData=[];
+var casesSizes=[];
+var casesTotals=[];
+var casesOutputs=[];
+
+function loadFileAsText() {
+    var fileToLoad = document.getElementById("fileToLoad").files[0];
+    var fileReader = new FileReader();
+    fileReader.onload = function(fileLoadedEvent) {
+        textLoaded = fileLoadedEvent.target.result;
+        divide(textLoaded);
+        loadCases();
+        searcher();
+    };
+    fileReader.readAsText(fileToLoad, "UTF-8");
+}
+
+function loadLocalFile() {
+    var xmlhttp = new XMLHttpRequest();
+    xmlhttp.onreadystatechange = function(){
+        if(xmlhttp.status == 200 && xmlhttp.readyState == 4){
+            textLoaded = xmlhttp.responseText;
+            divide(textLoaded);
```

Comment:
After loafing the file you do all your further actions inside a function called "loadLocalFile", this is not flexible enough. Code should be split down into small operations which do not really depend on each other (decoupled). So first youbload the file, then you take result and scan it with something, then you do search. We know that loading of file is an asynchronous operation. Sonyou can use callback to get result or promises.